features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,583 @@
+                            inputProps={{ min: "0" }}
+                            label={`Max collateral/token`}
+                            placeholder="1234"
+                            value={maxCollPerToken}
+                            onChange={(
+                              e: React.ChangeEvent<HTMLInputElement>
+                            ) => setMaxCollPerToken(e.target.value)}
+                          />
+                        </Grid>
+                        <Grid item xs={6}>
+                          <TextField
+                            fullWidth
+                            type="number"
+                            inputProps={{ min: "0" }}
+                            label={`Max tokens to liquidate`}
+                            placeholder="1234"
+                            error={tokenBalanceTooLow}
+                            helperText={
+                              tokenBalanceTooLow
+                                ? `${tokenSymbol} balance too low`
+                                : null
+                            }
+                            value={maxTokensToLiquidate}
+                            onChange={(
+                              e: React.ChangeEvent<HTMLInputElement>
+                            ) => setMaxTokensToLiquidate(e.target.value)}
+                          />
+                        </Grid>
+                        <Grid item xs={6}>
+                          <FormControl fullWidth>
+                            <InputLabel>Deadline</InputLabel>
+                            <Select
+                              value={deadline}
+                              onChange={(
+                                e: React.ChangeEvent<{ value: unknown }>
+                              ) => setDeadline(e.target.value as string)}
+                            >
+                              <MenuItem value={60}>1 minute</MenuItem>
+                              <MenuItem value={5 * 60}>5 minutes</MenuItem>
+                              <MenuItem value={30 * 60}>30 minutes</MenuItem>
+                              <MenuItem value={60 * 60}>1 hour</MenuItem>
+                              <MenuItem value={5 * 60 * 60}>5 hours</MenuItem>
+                              <MenuItem value={1911772800}>Forever</MenuItem>
+                            </Select>
+                          </FormControl>
+                        </Grid>
+                      </Grid>
+                    </Box>
+
+                    <Box textAlign="center" pt={2} pb={2}>
+                      {minCollPerToken &&
+                      maxCollPerToken &&
+                      maxTokensToLiquidate &&
+                      deadline &&
+                      !collBalanceTooLow() &&
+                      !tokenBalanceTooLow ? (
+                        <Button
+                          fullWidth
+                          variant="contained"
+                          onClick={executeLiquidation}
```

Comment:
This is where I think you can make better use of the `disabled={boolean}` prop and just disable this button unless all of the conditions are set. You decide not to render the button unless:

```
minCollPerToken &&
                      maxCollPerToken &&
                      maxTokensToLiquidate &&
                      deadline &&
                      !collBalanceTooLow() &&
                      !tokenBalanceTooLow ?
```

But I think its better UX to show the user the button in disabled mode. Also, you don't need to catch these conditions a second time in `executeLiquidation`

Reply:
Well, I actually think it's equivalent because in the case that those checks turn out to be false then I render a disabled button.

However, I agree with you that it's cleaner to put this in the `disable` field. rather than having this if/else logic. the only downside is that `disabled` is inherently the negative case so now the whole thing needs to be wrapped in a bracket like:
```
!(
                            minCollPerToken &&
                            maxCollPerToken &&
                            maxTokensToLiquidate &&
                            deadline &&
                            !collBalanceTooLow() &&
                            !tokenBalanceTooLow
                          )
```
which is arguably harder to read anyways.

Reply:
I just don't want to duplicate the checks and I don't see a problem showing a disabled button